src/app/game-state/home.service.ts

```diff
@@ -698,35 +699,52 @@ export class HomeService {
   }
 
   autoBuy(){
+    let priceBuffer = (this.home.costPerDay + 1) * 14; // Two weeks, by popular request.
     if (this.autoBuyHomeUnlocked && this.homeValue < this.autoBuyHomeLimit){
+      // check for final building to avoid buying excess building land.
+      if (!this.upgrading && this.nexthome.HomeType < this.autoBuyHomeLimit){
+        //try to buy land, and do it instantly.
+        while (this.characterService.characterState.money > this.landPrice + priceBuffer && this.land < this.nextHome.landRequired){
+          this.buyLand();
+        }
+      }
       if (this.land >= this.nextHome.landRequired){
-        //autoBuy is on, we have enough land, check if we have the money for the house plus the next few days' rent and food
-        if ((this.nextHome.cost + (this.nextHome.costPerDay * 3) + 3)  < this.characterService.characterState.money){
+        // autoBuy is on, we have enough land, check if we have the money for the house plus food and rent
+        if ((this.characterService.characterState.money > this.nextHome.cost + priceBuffer )){
           this.upgradeToNextHome();
         } else {
           // we can't afford the next house, bail out and don't autoBuy more land
           return;
         }
-      } else {
-        //try to buy land
-        if (this.characterService.characterState.money > this.landPrice  + (this.nextHome.costPerDay * 3) + 3 ){
-          this.buyLand();
-        }
       }
     }
-    if (this.autoBuyLandUnlocked && (this.land + this.fields.length + this.extraFields) < this.autoBuyLandLimit){
-      //autoBuy is on, check if we have the money for the land plus the next few days' rent and food
-      if (this.characterService.characterState.money > this.landPrice  + (this.nextHome.costPerDay * 3) + 3 ){
+    // if there's no autohome, autohome is finished, or there's enough money to cover buying a new plot of land and the home, try buying land.
+    if (!this.autoBuyHomeUnlocked ||
+        this.homeValue >= this.autoBuyHomeLimit ||
+        (this.characterService.characterState.money >= this.nextHome.cost + this.landPrice + priceBuffer )){
```

Comment:
Seems like we should just check the money in the while loop condition.

Reply:
It's an || statement, so without this check it won't buy anything if you have autohome and you haven't reached your buy limit. This makes sure you have more than enough for the next home and allows you to autobuy farmland during a build.